moxin-frontend/src/data/preferences.rs

```diff
@@ -7,12 +7,14 @@ use std::{
 use moxin_protocol::data::FileID;
 use serde::{Deserialize, Serialize};
 
-use super::filesystem::moxin_home_dir;
+use super::filesystem::{moxin_home_dir, setup_model_downloads_folder};
 const PREFERENCES_FILE: &str = "preferences.json";
 
 #[derive(Serialize, Deserialize, Debug, Clone, Default)]
 pub struct Preferences {
     pub current_chat_model: Option<FileID>,
+    #[serde(default = "setup_model_downloads_folder")]
```

Comment:
Is it good to use a function with side effects as a serde default?
Also, this function is currently blocking but would this still work if it becomes non-blocking?

Reply:
It's a good call. I'm reverting this